### DISCUSSION TITLE
How to grab HttpClientSendRequestTime from ExecutionInterceptor

### DISCUSSION THREAD

**Discussion by mahammadv** [aws/aws-sdk-java-v2#5533] [Migration from v1 to v2]
With v1, there was a way to extract information about HttpClientSendRequestTime, HttpClientReceiveResponseTime and HttpRequestTime which was helpful to figure out connectTime, request issue time, response receive time, etc. It looks like it might be possible to extract duration from beginning of the request till response received. But not exactly the time spent on issuing the request. Can someone help with this?

**Comment by debora-ito** [comment]
Can you use the SDK client-side metrics in V2?
https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/metrics.html

Or do you need to use ExecutionInterceptors?

**Reply by mahammadv** [reply]
The problem is that I would like to have it on a place where I get request and response information such as statusCode, statusText, etc along with this. This was fine with v1 where all these info was available.

**Reply by mahammadv** [reply]
With MetricsPublisher, I can't get request and response information all in the same place, correct?
